### PR TITLE
Create IsStar and replace chunk_is_star with it

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by dundargoc** [comment]
Tried to err on the side of caution and only skip adding `nullptr` if I'm absolutely sure it's not needed.

File: src/align_stack.cpp [hunk 1]
```diff
@@ -208,8 +208,9 @@ void AlignStack::Add(Chunk *start, size_t seqnum)
       // back up to the first '*' or '^' preceding the token
       Chunk *tmp_prev = ali->GetPrev();
 
-      while (  chunk_is_star(tmp_prev)
-            || chunk_is_msref(tmp_prev))
+      while (  tmp_prev != nullptr
```
Comment (micheleCTDE) [inline src/align_stack.cpp:211]: There is no need to test for nullptr here, tmp_prev will never be such.

File: src/align_stack.cpp [hunk 2]
```diff
@@ -296,14 +297,11 @@ void AlignStack::Add(Chunk *start, size_t seqnum)
...
 
-      if (  (  chunk_is_star(tmp)
-            && m_star_style == SS_DANGLE)
-         || (  chunk_is_addr(tmp)
-            && m_amp_style == SS_DANGLE)
-         || (  chunk_is_nullable(tmp)
-            && (m_star_style == SS_DANGLE))
-         || (  chunk_is_msref(tmp)
-            && m_star_style == SS_DANGLE)) // TODO: add m_msref_style
+      if (  tmp != nullptr
```
Comment (micheleCTDE) [inline src/align_stack.cpp:300]: There is no need to test for nullptr here, tmp_prev will never be such.

File: src/align_stack.cpp [hunk 3]
```diff
@@ -296,14 +297,11 @@ void AlignStack::Add(Chunk *start, size_t seqnum)
...
+         && (  (tmp->IsStar() && m_star_style == SS_DANGLE)
+            || (chunk_is_addr(tmp) && m_amp_style == SS_DANGLE)
+            || (chunk_is_nullable(tmp) && (m_star_style == SS_DANGLE))
+            || (chunk_is_msref(tmp) && m_star_style == SS_DANGLE))) // TODO: add m_msref_style
```
Comment (micheleCTDE) [inline src/align_stack.cpp:304]: The project maintainer (@guy-maurel) prefers to have the various conditions on separate lines.

File: src/combine.cpp [hunk 4]
```diff
@@ -1449,7 +1449,8 @@ void do_symbol_check(Chunk *prev, Chunk *pc, Chunk *next)
          }
          else if (  (  get_chunk_parent_type(pc) == CT_FUNC_DEF
                     && (  chunk_is_opening_brace(next)
-                       || chunk_is_star(pc->next)))
+                       || (  pc->GetNext() != nullptr
```
Comment (micheleCTDE) [inline src/combine.cpp:1452]: "pc->GetNext()" will never return a nullptr by design

File: src/combine.cpp [hunk 5]
```diff
@@ -3334,7 +3335,7 @@ static void handle_oc_message_send(Chunk *os)
    }
    else
    {
-      if (chunk_is_star(tmp)) // Issue #2722
+      if (tmp != nullptr && tmp->IsStar()) // Issue #2722
```
Comment (micheleCTDE) [inline src/combine.cpp:3338]: There is no need to test for nullptr here, tmp will never be such.

File: src/combine_fix_mark.cpp [hunk 6]
```diff
@@ -351,9 +351,10 @@ void fix_fcn_def_params(Chunk *start)
          continue;
       }
 
-      if (  chunk_is_star(pc)
-         || chunk_is_msref(pc)
-         || chunk_is_nullable(pc))
+      if (  pc != nullptr
```
Comment (micheleCTDE) [inline src/combine_fix_mark.cpp:354]: There is no need to test for nullptr here, pc will never be such.

File: src/combine_fix_mark.cpp [hunk 7]
```diff
@@ -1362,7 +1363,7 @@ void mark_function(Chunk *pc)
       tmp3 = chunk_get_next_ssq(tmp3);
 
       if (  chunk_is_str(tmp3, ")", 1)
-         && (  chunk_is_star(tmp1)
+         && (  (tmp1 != nullptr && tmp1->IsStar())
```
Comment (micheleCTDE) [inline src/combine_fix_mark.cpp:1366]: There is no need to test for nullptr here, tmp1 will never be such.

File: src/combine_fix_mark.cpp [hunk 8]
```diff
@@ -2181,9 +2182,9 @@ bool mark_function_type(Chunk *pc)
               get_token_name(tmp->type), tmp->Text(),
               tmp->orig_line, tmp->orig_col);
 
-      if (  chunk_is_star(tmp)
-         || chunk_is_token(tmp, CT_PTR_TYPE)
-         || chunk_is_token(tmp, CT_CARET))
+      if (tmp != nullptr && (  tmp->IsStar()
```
Comment (micheleCTDE) [inline src/combine_fix_mark.cpp:2185]: There is no need to test for nullptr here, tmp will never be such.

File: src/combine_fix_mark.cpp [hunk 9]
```diff
@@ -2499,7 +2500,8 @@ Chunk *mark_variable_definition(Chunk *start)
                  pcf_flags_str(pc->flags).c_str());
       }
       else if (  !bit_field_colon_is_present                      // Issue #2689
-              && (  chunk_is_star(pc)
+              && pc != nullptr
```
Comment (micheleCTDE) [inline src/combine_fix_mark.cpp:2503]: There is no need to test for nullptr here, pc will never be such.

**Review by micheleCTDE — CHANGES_REQUESTED** [review]
Please see comments for details

**Comment by micheleCTDE** [comment]
@dundargoc 
Additional note: as you may have noticed, most of the chunk_is_* functons have not been converted into Chunk::* methods yet. This is not a coincidence, but it is to minimize the amount of checking/rework related to nullptr.
The Chunk::Get* methods do not return a nullptr, so once all chunk_get_* functions have been converted, it should be much easier to convert the chunk_is* functions.

**Comment by dundargoc** [comment]
Aha, yes, that sounds much easier. Thanks for the tip.

**Comment by dundargoc** [comment]
Shall I squash?

**Comment by micheleCTDE** [comment]
> Shall I squash?

Would be good :-)

**Review by micheleCTDEAdmin — APPROVED** [review]

**Review by micheleCTDE — APPROVED** [review]

**Comment by micheleCTDE** [comment]
Thanks!

**Pull request merged**